modules/ting/src/Search/StatementGroupRender.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * @file
+ * The StatementGroupRender class.
+ */
+
+namespace Ting\Search;
+
+/**
+ * Class TingFilterGroupRendere
+ * Recursivly renders a statement group
+ *
+ * @package Ting\Search
+ */
+class StatementGroupRender {
+
+  // Provider-specific mapping, intialized during field rendering.
+  protected static $commonFieldMapping = NULL;
+
+  /**
+   * Recursively renders a statement group into a string.
+   *
+   * @param \Ting\Search\BooleanStatementGroup $group
+   *   The group to render.
+   *
+   * @return string
+   *   The rendered group, empty string if the group is empty.
+   *
+   * @throws \Exception
+   *   In case the group contains invalid members.
+   */
+  public static function renderGroup($group) {
+    return self::walk($group);
+  }
+
+  /**
+   * Renders a list of statements into a single string statement.
+   *
+   * This is a helper-function that warps the list in a single group before
+   * rendering it.
+   *
+   * @param \Ting\Search\BooleanStatementInterface[] $statements
+   *   The list of statements to render.
+   *
+   * @return string
+   *   The rendered group, empty string if the group is empty.
+   *
+   * @throws \Exception
+   *   In case the group contains invalid members.
+   */
+  public static function renderStatements($statements) {
+    return self::renderGroup(new BooleanStatementGroup($statements));
+  }
+
+  /**
+   * Recursively process the group.
+   *
+   * @param \Ting\Search\BooleanStatementGroup $group
+   *   The group to be rendered.
+   *
+   * @param string                             $rendered_statement
```

Comment:
Odd spacing here :smile:

Reply:
_really_ need to fix that phpstom setting :)